### PR TITLE
Rotating feature

### PR DESCRIPTION
**Pull request by vinhig**
Crow is a nice straightforward lib to draw 2D shapes. However, it lacks the opportunity to rotate them. So "be the change you want to see in the world blah blah". Here it goes!
Angle of rotation are passed through `DrawConfig` structure, are expressed in degrees and are `i32`. Rotation is applied in the vertex shader. An example with two new images is present (square and non-square dimension).
Tested on intel and nvidia hardware.

### CODE REVIEW

File: src/lib.rs
Comment (lcnr) [inline src/lib.rs:321]:
```suggestion
    ///
    /// Rotating by anything more precise than 90 steps can cause the source image
    /// to be slightly distorted, especially if it has a very low resolution.
    pub rotation: i32,
```
at least that's what I expect

**Review by lcnr — APPROVED** [review]
Hi, thanks for opening this PR :heart: 

I originally didn't add rotation as rotating an image causes it to not really be "pixel perfect" anymore.
I still see why one would want this and as the maintenance burden of this change is really low I think this is a good change.


Want to still look at this locally before merging and will only get to this after the 24th so I won't merge this until then. From what I can see this is already looking ready for merge.

What's the source for `textures/cat1.png` and `textures/cat2.png` and under which license were they published?

**Comment by vinhig** [comment]
`textures/cat1.png` and `textures/cat2.png` are in the `textures` folder. To be honest I didn't check its license (the non-square texture is actually a cropped version of the first), but it seems that it comes from [this article](https://www.wired.com/story/grumpy-cat-obit/). Maybe I should change them to images from wikimedia?

**Comment by lcnr** [comment]
> Maybe I should change them to images from wikimedia?

yeah, would be more comfortable that way :+1: 

(CI is failing because clippy also checks the tests, that might be more annoying than helpful tbh :laughing: )

**Comment by vinhig** [comment]
Image with [Unsplash license](https://unsplash.com/license). I never thought that finding a royalty free kitty image would be so difficult ^^.

**Comment by lcnr** [comment]
:laughing: gotta make your own

can you add a new file `textures/README.md` which credits the image source? After this that looks good to me

File: src/backend/state.rs
Comment (lcnr) [inline src/backend/state.rs:106]:
```suggestion
            gl::UniformMatrix2fv(uniforms.source_rotation, 1, gl::FALSE, rot_mat.as_ptr().cast::<f32>());
```
using `&array[0]` to get a pointer to the whole array is probably unsound, see https://rust-lang.zulipchat.com/#narrow/stream/136281-t-lang.2Fwg-unsafe-code-guidelines/topic/ptr.20to.20array.20using.20array.20indexing for more details.

File: src/backend/state.rs
Comment (lcnr) [inline src/backend/state.rs:325]:
```suggestion
                    rot_mat.as_ptr().cast(),
```
same as above

**Review by lcnr — COMMENTED** [review]
tried it out locally and looked over it once again.

Can you change the example to be framerate independent.

**Comment by vinhig** [comment]
License added.
Rotation is now pixel perfect and framerate independent in the example.

**Comment by vinhig** [comment]
And Merry Christmas !
(not so accessory, I should have started with that ^^)

**Comment by lcnr** [comment]
did some small adjustments :+1:

**Pull request merged**
